src/plugins/ucx/ucx_backend.cpp

```diff
@@ -336,52 +341,75 @@ class nixlUcxBackendH : public nixlBackendReqH {
 void nixlUcxEngine::progressFunc()
 {
     using namespace nixlTime;
-    pthrActive = 1;
 
     vramApplyCtx();
 
+    {
+        std::unique_lock<std::mutex> lock(pthrActiveLock);
+        pthrActive = true;
+    }
+    pthrActiveCV.notify_one();
+
+    std::vector<pollfd> poll_fds;
```

Comment:
consider using pre-created epoll fd

Reply:
i mean to use epoll_wait() instead of poll()

Reply:
Hmm, epoll_wait() would likely perform worse on small number of descriptors like we have here (it is not like this loop is going to handle 10k TCP sockets or something) and on top of that it is not UNIX spec and is not portable (so you won't be able to run NIXL without modifications on BSD if need be).